src/renderer/sw_engine/tvgSwShape.cpp

```diff
@@ -179,7 +179,7 @@ static void _dashCubicTo(SwDashStroke& dash, const Point* ctrl1, const Point* ct
     } else {
         while ((len - dash.curLen) > 0.0001f) {
             Bezier left, right;
-            if (dash.curLen > 0) {
+            if (dash.curLen > 0 || addZero) {
```

Comment:
Eventually, this is bad for performance, so we need to think about it carefully.

Reply:
I moved this part to 'else'.